lib/hooks.js

```diff
@@ -1,30 +1,41 @@
 const RiffError = require('./riff-error');
 const {types: errorTypes} = require('./errors');
 
-const handleError = (reject) => (err) => {
-    return reject(new RiffError(errorTypes.HOOK_RUNTIME, err));
+const handleError = (err) => {
+    if (err instanceof RiffError) {
+        return err
+    }
+    return new RiffError(errorTypes.HOOK_RUNTIME, err);
 };
 
-module.exports.guardWithTimeout = (hookFn, timeoutInMs) => {
-    return new Promise((resolve, reject) => {
-        const hookType = typeof hookFn;
-        if (hookType === 'undefined') {
-            return resolve();
-        }
-        if (hookType !== 'function') {
-            return reject(new RiffError(errorTypes.HOOK_INVALID, `Hooks must be functions, found: ${hookType}`));
-        }
+const invalidHookError = (hookType) => {
+    return new RiffError(errorTypes.HOOK_INVALID, `Hooks must be functions, found: ${hookType}`);
+};
+
+const timeoutError = (timeoutInMs) => {
+    return new RiffError(errorTypes.HOOK_TIMEOUT, `The hook took too long to run (timeout: ${timeoutInMs}ms). Aborting now`);
+};
+
+module.exports.guardWithTimeout = async (hookFn, timeoutInMs) => {
+    const hookType = typeof hookFn;
```

Comment:
it's clearer (and counter intuitively faster) to not capture the type as a variable, instead do the typeof call directly in each condition expression.